Automattic/woocommerce-payments
# Remove setConfig()

Fixes #1330

#### Changes proposed in this Pull Request

Since wcSettings.setSetting is deprecrated and no longer available, remove `setConfig()` completely.
There is only one place where `setConfig()` is used where it was intended to set nonce from server to a global variable.
Fortunately, in that only 1 case, we can just set nonce to a variable as it's still with in the same context where nonce is needed in ajax call.

#### Testing instructions

* Install [WooCommerce Blocks](https://wordpress.org/plugins/woo-gutenberg-products-block/).
* Create a new page with the checkout block. I have 2 separate checkout pages: one for shortcode checkout and one for checkout block.
<img width="416" alt="Screen Shot 2021-04-14 at 12 17 49" src="https://user-images.githubusercontent.com/73803630/114658140-85d5ef00-9d1b-11eb-8582-4e57b8d3b23e.png">

* Add a product to cart and checkout with checkout block.
* Use SCA test credit card `4000002500003155`.
* With base branch, expect an error: `wc.wcSettings.setSetting is not a function`.
* With this branch, expect success.
* Test the same flow with shortcode checkout because the code change affects shortcode checkout as well.
* Test guest checkout (in incognito mode maybe in case you're logged in as admin) and choose to create an account during checkout. Make sure `Allow customers to create an account during checkout` is enabled in wp-admin > WooCommerce > Settings > Accounts & Privacy.
This can only be tested with shortcode checkout because I don't see checkout block lets you create an account.
We need to test this because of [this comment](https://github.com/Automattic/woocommerce-payments/blob/develop/client/checkout/api/index.js#L159-L160) even though I find that the code will always pass nonce, not only for that specific flow.

-------------------
